### PR TITLE
Add e2e-runner and e2e-jobs stub.

### PR DESCRIPTION
**Pull request by spxtr**
Step one of #18981 on master branch.

### CODE REVIEW

File: hack/jenkins/job-configs/kubernetes-e2e.yaml [hunk 1]
```diff
@@ -5,7 +5,8 @@
         daysToKeep: 7
     builders:
         - shell: |
-            curl -fsS --retry 3  "https://raw.githubusercontent.com/kubernetes/kubernetes/{branch}/hack/jenkins/e2e.sh" | bash -
+            curl -fsS --retry 3  "https://raw.githubusercontent.com/kubernetes/kubernetes/{branch}/hack/jenkins/e2e-jobs.sh" | bash -
+            curl -fsS --retry 3  "https://raw.githubusercontent.com/kubernetes/kubernetes/{branch}/hack/jenkins/e2e-runner.sh" | bash -
```
Comment (ixdy) [inline hack/jenkins/job-configs/kubernetes-e2e.yaml]: I think we want to use `master` here, so we don't need to update the runner in 3 places?

File: hack/jenkins/job-configs/kubernetes-e2e.yaml [hunk 2]
```diff
@@ -5,7 +5,8 @@
         daysToKeep: 7
     builders:
         - shell: |
-            curl -fsS --retry 3  "https://raw.githubusercontent.com/kubernetes/kubernetes/{branch}/hack/jenkins/e2e.sh" | bash -
+            curl -fsS --retry 3  "https://raw.githubusercontent.com/kubernetes/kubernetes/{branch}/hack/jenkins/e2e-jobs.sh" | bash -
```
Comment (ixdy) [inline hack/jenkins/job-configs/kubernetes-e2e.yaml]: hm. do environment variables that get exported here persist into the next script?

File: hack/jenkins/job-configs/kubernetes-e2e.yaml (hunk 1 above)
Comment (spxtr) [inline hack/jenkins/job-configs/kubernetes-e2e.yaml]: That's right.

File: hack/jenkins/job-configs/kubernetes-e2e.yaml (hunk 2 above)
Comment (spxtr) [inline hack/jenkins/job-configs/kubernetes-e2e.yaml]: I can source them instead.

**Comment by spxtr** [comment]
I decided to pull the yaml update to a second PR, so that I can test it out without (possibly) breaking every job it manages :P

**Comment by spxtr** [comment]
@ixdy PTAL

**Comment by ixdy** [comment]
LGTM

**Comment by ixdy** [comment]
Some sort of comment at the top describing these would be good in the next iteration.

**Pull request merged**